server/store.go

```diff
@@ -107,6 +107,18 @@ func (s *storeInfo) getLabelValue(key string) string {
 	return ""
 }
 
+func (s *storeInfo) getLocationID(keys []string) string {
+	id := ""
+	for _, k := range keys {
+		v := s.getLabelValue(k)
+		if len(v) == 0 {
+			return ""
```

Comment:
I don't think we should support missing label cases, it's error prone and hard to explain the behavior.
If we want to update the configurations or labels, we can just turn off the balance temporarily.